core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -105,10 +107,20 @@ class SparkHadoopWriter(@transient jobConf: JobConf)
   def commit() {
     val taCtxt = getTaskContext()
     val cmtr = getOutputCommitter()
+    val dagSchedulerActor =
+      AkkaUtils.makeDriverRef("DAGScheduler", SparkEnv.get.conf, SparkEnv.get.actorSystem)
+    val askTimeout = AkkaUtils.askTimeout(SparkEnv.get.conf)
     if (cmtr.needsTaskCommit(taCtxt)) {
       try {
-        cmtr.commitTask(taCtxt)
-        logInfo (taID + ": Committed")
+        val canCommit: Boolean = AkkaUtils.askWithReply(
+          AskPermissionToCommitOutput(jobID, splitID, attemptID), dagSchedulerActor, askTimeout)
```

Comment:
You can't know in advance which tasks might have speculated copies, so I think you could only enable/disable this depending on whether speculation was enabled rather than on a per-task or job basis.  Note that this only affects ResultTasks that save output to Hadoop, such as `saveAsTextFile`.